src/languageservice/jsonSchema.ts

```diff
@@ -90,3 +90,7 @@ export interface JSONSchema {
 export interface JSONSchemaMap {
   [name: string]: JSONSchemaRef;
 }
+
+export function isJSONSchema(schema: JSONSchemaRef): schema is JSONSchema {
+  return typeof schema !== 'boolean';
```

Comment:
This check does not make sense to me. Maybe you can explain the idea of what you are trying to do.

Could `schema` ever be a boolean? It looks like it would be either be `JSONSchemaRef` or `undefined`. 

It looks like this might be a type-guard function. If so then maybe something like this would work:
```
export function isJSONSchema(schema: any): schema is JSONSchema {
  return schema instanceof JSONSchema;
}
```

Reply:
I can see below that this is indeed a type-guard. I do not know if the `JSONSchemaRef` is actually a boolean, I'm guessing it is not. If it is not then checking that it is not a boolean will show you nothing. It will always be true.

You need unit tests that tests both sides of this.

Reply:
The JSONSchemaRef type is: `export type JSONSchemaRef = JSONSchema | boolean;`

Reply:
`isJSONSchema` is a type-guard function.
`JSONSchema` is interface, I cannot use `instanceof` to check  that `schema` type is `JSONSchema`

Reply:
This function is saying, if you check that it is NOT boolean then as long as it is anything but a boolean then it must be `JSONSchema`. Logically that does not make sense.

Reply:
But we only two choices, `schema`  can be boolean or `JSONSchema`
It would be enough to check that  `schema` is not boolean, as rest check will do TS compiler, it will not allow to pass anything to `isJSONSchema` which not looks like boolean or  `JSONSchema`.